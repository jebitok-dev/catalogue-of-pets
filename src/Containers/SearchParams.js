import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { ANIMALS } from '@frontendmasters/pet';
import Results from '../Components/Results';
import petFinder from '../Helpers/Api';

const SearchParams = ({
  pets = [], fetchPets, filterPets, filter,
}) => {
  const [petFilter, updateFilter] = useState(filter);
  const [setPets, updatePets] = useState(pets || []);

  const capitalize = ([first, ...rest]) =>
        first.toUpperCase() + rest.join('').toLowerCase(); // eslint-disable-line

  async function requestPets() {
    const { animals } = await petFinder.animal
      .search({
        type: petFilter,
      })
      .then((data) => data.data)
      .catch((error) => error);

    fetchPets(animals);
    updatePets(animals);
    filterPets(petFilter);
  }

  useEffect(async () => {
    const { animals } = await petFinder.animal
      .search({
        type: '',
        size: '',
        gender: '',
      })
      .then((data) => data.data);

    fetchPets(animals);
    updatePets(animals);
  }, []);

  return (
    <div className="search-params" data-testid="search-params-id">
      <form
        data-testid="search-params-form"
        className="search-params-form"
        onSubmit={(e) => {
          e.preventDefault();
          requestPets();
        }}
      >
        <label htmlFor="animal">
          Animal
          <select
            value={petFilter}
            onChange={(e) => updateFilter(e.target.value)}
          >
            <option disabled hidden defaultValue="">Select</option>
            {ANIMALS.map((option) => (
              <option
                value={option}
                key={option}
              >
                {capitalize(option)}
              </option>
            ))}
          </select>
        </label>
        <button type="submit">Submit</button>
      </form>
      <Results pets={setPets} />
    </div>
  );
};

SearchParams.propTypes = {
  pets: PropTypes.array, // eslint-disable-line
  fetchPets: PropTypes.func,
  filter: PropTypes.string,
  filterPets: PropTypes.func,
};

SearchParams.defaultProps = {
  pets: [],
  fetchPets: null,
  filter: 'All',
  filterPets: null,
};

export default SearchParams;
