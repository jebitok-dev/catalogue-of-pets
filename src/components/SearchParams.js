import React, { useState, useEffect } from 'react';
/* eslint-disable */
import pet, { ANIMALS } from '@frontendmasters/pet';
import Dropdown from './Dropdown';
import Results from './Results';

const SearchParams = () => {
  const [location, updateLocation] = useState('Seattle, WA');
  const [breeds, updateBreeds] = useState([]);
  const [animal, AnimalDropDown] = Dropdown('Animal', 'dog', ANIMALS);
  const [breed, BreedDropDown, updateBreed] = Dropdown('Breed', '', breeds);
  const [Pets, setPets] = useState([]);

  async function requestPets() {
    const { animals } = await pet.animals({
      location,
      breed,
      type: animal,
    });

    setPets(animals || []);
  }

  useEffect(() => {
    updateBreeds([]);
    updateBreed('');
    pet.breeds(animal).then(({ breeds }) => {
      const breedItems = breeds.map(({ name }) => name);
      updateBreeds(breedItems);
    });
  }, [animal, updateBreed, updateBreeds]);

  return (
    <div className="search-params">
      {name}
      <form onSubmit={(e) => {
        e.preventDefault();
        requestPets();
      }}
      >
        <label
          htmlFor="location"
        >
          location
          <input
            id="location"
            value={location}
            onChange={(e) => updateLocation(e.target.value)}
          />
        </label>
        <AnimalDropDown />
        <BreedDropDown />
        <button
          type="submit"
        >
          Submit
        </button>
      </form>
      <Results pets={Pets} />
    </div>
  );
};

export default SearchParams;
