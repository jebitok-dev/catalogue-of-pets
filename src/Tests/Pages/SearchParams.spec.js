import React from 'react';
import renderer from 'react-test-renderer';
import SearchParams from '../../Containers/SearchParams';

test('should correctly render SearchParams', () => {
  const component = renderer.create(<SearchParams />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
