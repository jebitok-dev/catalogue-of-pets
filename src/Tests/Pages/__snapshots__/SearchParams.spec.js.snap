// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render SearchParams 1`] = `
<div
  className="search-params"
  data-testid="search-params-id"
>
  <form
    className="search-params-form"
    data-testid="search-params-form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="animal"
    >
      Animal
      <select
        onChange={[Function]}
        value="All"
      >
        <option>
          Select
        </option>
        <option
          value="barnyard"
        >
          Barnyard
        </option>
        <option
          value="bird"
        >
          Bird
        </option>
        <option
          value="cat"
        >
          Cat
        </option>
        <option
          value="dog"
        >
          Dog
        </option>
        <option
          value="horse"
        >
          Horse
        </option>
        <option
          value="rabbit"
        >
          Rabbit
        </option>
        <option
          value="scales-fins-other"
        >
          Scales-fins-other
        </option>
        <option
          value="small-furry"
        >
          Small-furry
        </option>
      </select>
    </label>
    <label
      htmlFor="size"
    >
      Size
      <select
        onChange={[Function]}
        value="All"
      >
        <option>
          Select
        </option>
        <option
          value="Small"
        >
          Small
        </option>
        <option
          value="Medium"
        >
          Medium
        </option>
        <option
          value="Large"
        >
          Large
        </option>
      </select>
    </label>
    <label
      htmlFor="gender"
    >
      Gender
      <select
        onChange={[Function]}
        value="All"
      >
        <option>
          Select
        </option>
        <option
          value="Male"
        >
          Male
        </option>
        <option
          value="Female"
        >
          Female
        </option>
      </select>
    </label>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
  <div
    className="search"
  >
    <h1>
      No Pets Found
    </h1>
  </div>
</div>
`;
